sieve/script.c

```diff
@@ -841,6 +842,19 @@ static int do_action_list(sieve_interp_t *interp,
                          ACTIONS_STRING_LEN-strlen(actions_string),
                          "Discarded\n");
             break;
+        case ACTION_IMIP:
+            if (interp->imip)
+                ret = interp->imip(NULL, interp->interp_context,
+                                   script_context,
+                                   message_context,
+                                   &errmsg);
+            free(interp->lastitem);
+            interp->lastitem = NULL;
+            if (ret == SIEVE_OK)
```

Comment:
If we somehow get here without an `interp->imip` callback, then we won't set `ret` ourselves.  And if the previous action had been successful, then we'll misreport this one as success too, even though we did nothing.

I notice the ACTION_VACATION case below returns `SIEVE_INTERNAL_ERROR` if there's no callback, should this do the same?

Reply:
Fixed here: https://github.com/cyrusimap/cyrus-imapd/pull/3699/commits/5bf165415498497bf4e2f81fe3a9a8b50dbdd104